### ISSUE TITLE
Add action for GitHub release

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-web#304]
(related to #303)

I'm introducing an action like
```yml
name: Create GitHub release
on:
  push:
    tags:
      - '[0-9]+.[0-9]+.[0-9]+'
      - '[0-9]+.[0-9]+.[0-9]+[a-c][0-9]+'
      - '[0-9]+.[0-9]+.[0-9]+alpha[0-9]+'
      - '[0-9]+.[0-9]+.[0-9]+beta[0-9]+'
      - '[0-9]+.[0-9]+.[0-9]+rc[0-9]+'

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Release
        uses: softprops/action-gh-release@v1
```

With this action in place, any time we create a new tag (and if it matches with the typical format of a version), a new GitHub release is created.

For the moment I'm not adding any automated-release-notes feature, and then we will have to edit the release by hand.

---

TLDR, after this issue and #303 are both closed, the new procedure for releasing will be like this:
* Current version is 0.1.2
* Make any number of PRs, and update CHANGELOG
* Choose new version string (e.g 0.2.0)
* Update version in the file that hardcodes it (still undefined, see #303), and push this change
* Push the tag 0.2.0
